src/com/google/javascript/jscomp/PrintStreamJSONErrorManager.java

```diff
@@ -0,0 +1,158 @@
+        int charno = message.error.getCharno();
+
+        jsonWriter.beginObject();
+        jsonWriter.name("level").value(message.level == CheckLevel.ERROR ? "error" : "warning");
+        jsonWriter.name("description").value(message.error.description);
+        jsonWriter.name("source").value(sourceName);
+        jsonWriter.name("line").value(lineNumber);
+        jsonWriter.name("column").value(charno);
+
+        // extract source excerpt
+        String sourceExcerpt = SourceExcerpt.LINE.get(
+            sourceExcerptProvider, sourceName, lineNumber, excerptFormatter);
+        if (sourceExcerpt != null) {
+          StringBuilder b = new StringBuilder(sourceExcerpt);
+          b.append("\n");
+
+          // padding equal to the excerpt and arrow at the end
+          // charno == sourceExcerpt.length() means something is missing
+          // at the end of the line
+          if (0 <= charno && charno <= sourceExcerpt.length()) {
+            for (int i = 0; i < charno; i++) {
+              char c = sourceExcerpt.charAt(i);
+              if (TokenUtil.isWhitespace(c)) {
+                b.append(c);
+              } else {
+                b.append(' ');
+              }
+            }
+            if (message.error.node == null) {
+              b.append("^");
+            } else {
+              int length = Math.max(
+                  1, Math.min(message.error.node.getLength(), sourceExcerpt.length() - charno));
+              for (int i = 0; i < length; i++) {
+                b.append("^");
+              }
+            }
+          }
+
+          jsonWriter.name("context").value(b.toString());
+        }
+
+        OriginalMapping mapping = sourceExcerptProvider.getSourceMapping(
+            sourceName, message.error.lineNumber, message.error.getCharno());
+
+        if (mapping != null) {
+          jsonWriter.name("originalLocation").beginObject();
+          jsonWriter.name("source").value(mapping.getOriginalFile());
+          jsonWriter.name("line").value(mapping.getLineNumber());
+          jsonWriter.name("column").value(mapping.getColumnPosition());
+          jsonWriter.endObject();
+        }
+
+        jsonWriter.endObject();
+        list.add(message);
+      }
+
+
+      StringBuilder summaryBuilder = new StringBuilder();
+      if (getTypedPercent() > 0.0) {
+        summaryBuilder.append(SimpleFormat.format("%d error(s), %d warning(s), %.1f%% typed",
+            getErrorCount(), getWarningCount(), getTypedPercent()));
+      } else {
+        summaryBuilder.append(SimpleFormat.format("%d error(s), %d warning(s)", getErrorCount(),
+            getWarningCount()));
+      }
+      jsonWriter.beginObject();
+      jsonWriter.name("level").value("info");
+      jsonWriter.name("description").value(summaryBuilder.toString());
+      jsonWriter.endObject();
+
+      jsonWriter.endArray();
+      jsonWriter.flush();
+      jsonWriter.close();
+    } catch (UnsupportedEncodingException e) {
+
```

Comment:
I believe you should re-throw the caught exception wrapped in a `RuntimeException` so errors aren't just swallowed.